### PR TITLE
fix(minify): upgrade terser to v5, etc.

### PR DESCRIPTION
**Pull request by Francois-Esquire**
This PR upgrades `terser` to v5 and adds a test case to test for ES2020 syntax (specifically optional chaining). Relates to issue #17 and #19.

### CODE REVIEW

**Comment by brody2consult** [comment]
> adds a test case to test for ES2020 syntax

It would be awesome if you want to put this info into the commit message.

Unfortunately #18 (remove Node.js pre-14 support) seems to be a prerequisite for this. I may do this soon.

I may have a few more nits coming on this, and some more comments coming on original issue #17.

---

P.S. Adding a link to related issue: https://github.com/TrySound/rollup-plugin-size-snapshot/issues/61

File: README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ If you use uglify or terser plugins, then make sure they are placed after this o
 
 ```js
 import { uglify } from "rollup-plugin-uglify";
-// or import { terser } from "rollup-plugin-terser";
+// or import terser from "@rollup/plugin-terser";
```
Comment (brody2consult) [inline README.md:60]: Yup thanks!

File: src/index.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ export const sizeSnapshot = (options?: Options = {}): Plugin => {
       // brodybits/rollup-plugin-size-snapshot#17
       // but a reproduction is needed to add a test case, see
       // https://github.com/brodybits/rollup-plugin-size-snapshot/issues/19
-      const minified = minify(source).code;
+      const minified = (await minify(source)).code;
```
Comment (brody2consult) [inline src/index.js]: OK for now, Terser error handling still looks like a loose end as I said in https://github.com/brodybits/rollup-plugin-size-snapshot/issues/17#issuecomment-1402324621

File: tests/fixtures/next-features.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+export function Component() {
+  const ref = React.useRef();
+
+  React.useEffect(() => {
+    ref.current?.focus();
+  });
+
+  return React.createElement("div", { ref });
+}
```
Comment (brody2consult) [inline tests/fixtures/next-features.js]: I would prefer that we rename this to something like es2020-features.js as I would interpret "next" as referring to the future.

File: tests/index.test.js [hunk 4]
```diff
@@ -123,6 +123,24 @@ test("not affected by following terser plugin", async () => {
   });
 });
 
+test("minifies with latest ES syntax", async () => {
+  const snapshotPath = "fixtures/next-features.size-snapshot.json";
```
Comment (brody2consult) [inline tests/index.test.js]:
```suggestion
test("minifies with ES2020 syntax", async () => {
  const snapshotPath = "fixtures/es2020-features.size-snapshot.json";
```

File: tests/index.test.js [hunk 5]
```diff
@@ -123,6 +123,24 @@ test("not affected by following terser plugin", async () => {
...
+  await runRollup({
+    external: ["react"],
+    input: "./fixtures/next-features.js",
+    output: { file: "fixtures/next-features.esm.js", format: "esm" },
```
Comment (brody2consult) [inline tests/index.test.js]:
```suggestion
    input: "./fixtures/es2020-features.js",
    output: { file: "fixtures/es2020-features.esm.js", format: "esm" },
```

**Review by brody2consult — COMMENTED** [review]
Looks good in general.

My one major criticism is the use of "next" as ES2020 would be 2-3 years old by now.

IMHO adding tests is not as important when updating dependencies. I think tests are way more important when making bug fixes or adding new features.

It looks to me like this proposal includes several related items, which is why I added ", etc." to the title of this PR:

- update Terser to v5
- add test of ES2020
- test with updated @rollup/plugin-terser
- update README.md with updated @rollup/plugin-terser

**Comment by brody2consult** [comment]
@Francois-Esquire I have just released another patch and started version 0.14.0 with minimum Node.js version 14. I would like to get the terser v5 upgrade out pretty soon. If you can rebase this and address my comments soon then I can merge it in. Otherwise I may just do this upgrade without the test case.

I will start updating some other dependencies now.

**Comment by brody2consult** [comment]
@Francois-Esquire never mind, I am taking the liberty to fix this up myself and will merge this as a squash commit. I would like to get this finished soon. Thanks and apologies.

**Comment by Francois-Esquire** [comment]
@brodybits Thanks so much for taking the lead on this, truly appreciate it! And thanks for staying committed to this package, has been a huge help for us 🙌

**Comment by brody2consult** [comment]
These updates are now published in v0.14.1 (0.14.0 was published with missing contents).

FYI I am currently available for short-term and long-term opportunities in case anyone in your organization may be interested, private contact <chris.brody@gmail.com>.

**Pull request merged**